File: kinotic-management-api/src/main/java/org/kinotic/management/api/model/UiDeployment.java
package org.kinotic.management.api.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.kinotic.domain.api.model.DeploymentStatus;
import org.kinotic.domain.api.model.OrganizationScoped;

import java.util.Date;

/**
 * The standing deployment of one UI artifact of a {@link Project}: the site serving it, the
 * commit it serves, and its status. One row per UI a deployment has published; a row outlives
 * the artifact until the deployment is removed.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
public class UiDeployment implements OrganizationScoped<String> {

    /**
     * The site's hostname label under the platform's sites domain, minted once when the UI
     * is first published and never parsed: {@code <org>-<app>-<ui>}, with a numeric suffix
     * when that label is taken. The site's resources are named after it.
     */
    private String id;

    private String organizationId;

    private String applicationId;

    /**
     * The id of the project the UI belongs to.
     */
    private String projectId;

    /**
     * The UI's identity: the {@link UiArtifact#name()} it was published from. Unique among
     * the project's UI deployments.
     */
    private String name;

    /**
     * Where the site is served, {@code https://<id>.<sites domain>}, fixed when the label is
     * minted.
     */
    private String url;

    /**
     * Sha of the commit the site serves, or {@code null} until the first publish completes.
     */
    private String commitSha;

    private DeploymentStatus status;

    private Date created;

    private Date updated;

}

File: kinotic-js/workspace/packages/management-api/src/api/model/UiDeployment.ts
import type { Identifiable } from '@kinotic-ai/core'
import type { DeploymentStatus } from '@/api/model/DeploymentStatus'

/**
 * The standing deployment of one UI artifact of a Project: the site serving it, the commit
 * it serves, and its status. One row per UI a deployment has published; a row outlives the
 * artifact until the deployment is removed.
 */
export class UiDeployment implements Identifiable<string> {

    /**
     * The site's hostname label under the platform's sites domain, minted once when the UI
     * is first published: `<org>-<app>-<ui>`, with a numeric suffix when that label is taken.
     */
    public id: string | null = null

    public organizationId!: string

    public applicationId!: string

    /**
     * The id of the project the UI belongs to.
     */
    public projectId!: string

    /**
     * The UI's identity: the UiArtifact name it was published from. Unique among the
     * project's UI deployments.
     */
    public name!: string

    /**
     * Where the site is served, `https://<id>.<sites domain>`, fixed when the label is minted.
     */
    public url!: string

    /**
     * Sha of the commit the site serves, or null until the first publish completes.
     */
    public commitSha: string | null = null

    public status!: DeploymentStatus

    public created: number | null = null

    public updated: number | null = null

}
